pkg/api/devportal/ui.go

```diff
@@ -103,10 +106,23 @@ func templatePortalIndexes(indexTemplate *template.Template, portals []portal) (
 		for _, customDomain := range p.Status.CustomDomains {
 			indexes[customDomain] = templated
 		}
-		if len(p.Status.CustomDomains) == 0 {
-			indexes[p.Status.HubDomain] = templated
-		}
+		indexes[p.Status.HubDomain] = templated
 	}
 
 	return indexes, nil
 }
+
+// stripHostPort returns host without any trailing ":<port>".
+// https://github.com/golang/go/blob/cdf77c7209a497825b2956ec0360c6e7e4ae0acd/src/net/http/server.go#L2358-L2368
+func stripHostPort(host string) string {
+	// If no port on host, return unchanged
+	if !strings.Contains(host, ":") {
+		return host
+	}
+
+	host, _, err := net.SplitHostPort(host)
+	if err != nil {
+		return host // on error, return unchanged
+	}
+	return host
```

Comment:
```suggestion
func stripHostPort(h string) string {
	// If no port on host, return unchanged
	if !strings.Contains(h, ":") {
		return host
	}

	host, _, err := net.SplitHostPort(h)
	if err != nil {
		return h // on error, return unchanged
	}
	return host
```

This would fix the error handling of `net.SplitHostPort`